pkg/storage/postgres/postgres.go

```diff
@@ -37,6 +38,38 @@ type Postgres struct {
 var _ storage.OpenFGADatastore = (*Postgres)(nil)
 
 func New(uri string, cfg *sqlcommon.Config) (*Postgres, error) {
+
+	// Is username or password explicitly provided?
+	if cfg.Username != "" || cfg.Password != "" {
+		parsed, err := url.Parse(uri)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse postgres connection uri: %w", err)
+		}
+
+		// Username always set to some value
+		username := ""
+		if cfg.Username != "" {
+			username = cfg.Username
+		} else if parsed.User != nil {
+			username = parsed.User.Username()
+		}
+
```

Comment:
Do we need to do anything if username is still ""?

Reply:
Username will still be "" in the following cases, for example:

`postgres://:password@localhost:5432/postgres`
`postgres://localhost:5432/postgres`

```shell
export OPENFGA_DATASTORE_USERNAME=""

./openfga run --datastore-engine mysql --datastore-uri 'root:secret@tcp(localhost:3306)/openfga?parseTime=true'
```

In these cases we constrct the credentials correctly, and we attempt to connect with a username that is an empty string, which may or may not work depending on the user credentials. In any case though, the behavior seems to override correctly and accurately based on what is provided.

Reply:
Thanks for the explanation.  In this case, the comment `// Username always set to some value` is a bit misleading

Reply:
Gotcha. Those comments are a bit unecessary. The code is pretty self-explanatory, so I just removed them.